packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -247,10 +247,10 @@ exports.sourceNodes = async (
         if (node.internal.owner !== `gatsby-source-contentful`) {
           return
         }
-        touchNode({ nodeId: node.id })
+        touchNode(node)
         if (node.localFile___NODE) {
           // Prevent GraphQL type inference from crashing on this property
-          touchNode({ nodeId: node.localFile___NODE })
+          touchNode({ ...node, id: node.localFile___NODE })
```

Comment:
Could you update to use the getNode syntax
```suggestion
          touchNode(getNode(node.localFile___NODE )
```